executors/clipseg/executor.py

```diff
@@ -0,0 +1,214 @@
+import enum
+import os
+import sys
+import shutil
+import time
+
+from PIL import Image, ImageOps
+from io import BytesIO
+from pathlib import Path
+from typing import Dict
+from urllib.request import urlopen
+
+import cv2
+import numpy as np
+import torch
+
+from jina import Executor, DocumentArray, Document, requests
+from torchvision import transforms
+
+
+class THRESHOLDING_METHODS(str, enum.Enum):
+    NONE = 'none' # Do not threshold
+    BINARY = 'binary'
+    ADAPTIVE_MEAN = 'adaptive_mean'
+    ADAPTIVE_GAUSSIAN = 'adaptive_gaussian'
+
+
+DALLE_CACHE_DOCKER_DIR = '/home/dalle/.cache'
+THRESHOLD_ADAPTIVE_DEFAULT_BLOCK_SIZE = 11
+THRESHOLD_ADAPTIVE_DEFAULT_C = 2.
+THRESHOLD_BINARY_DEFAULT_STRENGTH_VALUE = 85
+SELF_DIR = os.path.dirname(os.path.abspath(__file__))
+WEIGHTS_LOC = 'https://owncloud.gwdg.de/index.php/s/ioHbRzFx6th32hn/download'
+WEIGHT_FOLDER_NAME = 'clipseg_weights'
+WEIGHT_ZIP_FILE_NAME = 'clipseg_weights.zip'
+
+TRANSFORM_FOR_SEGMENTATION = transforms.Compose([
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    transforms.Resize((512, 512)),
+])
+
+
+def document_to_pil(doc):
+    uri_data = urlopen(doc.uri)
+    return Image.open(BytesIO(uri_data.read()))
```

Comment:
lets have that as static method as well. And please add type hint for doc and the return type